File: src/main/java/net/azisaba/townia/data/TownRank.java
package net.azisaba.townia.data;

public enum TownRank {
    RESIDENT(0),
    ASSISTANT(1),
    CO_MAYOR(2),
    MAYOR(3);

    private final int level;

    TownRank(int level) {
        this.level = level;
    }

    public int getLevel() {
        return level;
    }

    public boolean isAtLeast(TownRank other) {
        return this.level >= other.level;
    }

    public static TownRank fromString(String s) {
        try {
            return TownRank.valueOf(s.toUpperCase());
        } catch (IllegalArgumentException e) {
            return RESIDENT;
        }
    }
}

File: src/main/java/net/azisaba/townia/data/TownRank.kt
package net.azisaba.townia.data

import java.util.*

enum class TownRank(val level: Int) {
    RESIDENT(0),
    ASSISTANT(1),
    CO_MAYOR(2),
    MAYOR(3);

    fun isAtLeast(other: TownRank): Boolean {
        return this.level >= other.level
    }

    companion object {
        fun fromString(s: String): TownRank {
            try {
                return valueOf(s.uppercase(Locale.getDefault()))
            } catch (e: IllegalArgumentException) {
                return TownRank.RESIDENT
            }
        }
    }
}